### ISSUE TITLE
Toggle checkboxes in view mode

### ISSUE THREAD

**Issue by stefan-niedermann** [stefan-niedermann/nextcloud-notes#451]
After #422 it is now possible to implement support for toggling checkboxes in the view mode.

There is already commented code, that does not completely work.

**Comment by toxpal** [comment]
Any update on this feature? It's the only feature I miss badly about the app.

**Comment by sscholl** [comment]
+1 thanks for your work!

**Comment by poikilos** [comment]
@stefan-niedermann do you use the same markdown libary as the official (iOS apparently) version (take a look at this code: <https://github.com/nextcloud/notes/pull/303>)?

**Comment by tobru** [comment]
Was the bug solved lately in the upstream library? If not, would you mind to link the bugreport here, so someone might be able to help with fixing.

**Comment by stefan-niedermann** [comment]
Sure, this was the core problem when i tried to implement this feature: https://github.com/yydcdut/RxMarkdown/issues/55

Basically everytime when i clicked the checkbox, the markdown was presented as plain text after clicking, and not rendered.

This is also the problem for another feature (refresh in view mode). Every help is very appreciated :)

**Comment by tobru** [comment]
Thanks for the pointer. Looks like the RxMarkdown projects isn't really active anymore since a long time. So IMHO this issue relates to #549 (or a follow-up issue to replace the markdown engine)

**Comment by stefan-niedermann** [comment]
Yes, that's right. I am looking for alternatives, but as long as we don't have one, we are going with RxMarkdown. I will have a look at markor, as soon as i have time.

Though this does not really help for the issue. I will try to get in contact with the developer of RxMarkdown. I think (s)he might have not much time to develop, but if someone starts a PR, maybe it will get merged.

**Comment by tobru** [comment]
Thanks @stefan-niedermann for taking care :+1:

**Comment by quthla** [comment]
> Yes, that's right. I am looking for alternatives, but as long as we don't have one, we are going with RxMarkdown. I will have a look at markor, as soon as i have time.
> 
> Though this does not really help for the issue. I will try to get in contact with the developer of RxMarkdown. I think (s)he might have not much time to develop, but if someone starts a PR, maybe it will get merged.

https://github.com/stefan-niedermann/nextcloud-notes/issues/549#issuecomment-513489407

How about this alternative? The editor seems pretty solid by a first test

**Comment by panki27** [comment]
Hey, any update on this yet? 
If I spam the checkbox in view mode (v 1.0.1) I get an 
`IllegalArgumentException: "Center point is not inside any of the rectangles!"`
So.. it's "half implemented"?

**Comment by stefan-niedermann** [comment]
I am currently busy with Single Sign On and Multi Account, but this is on my list definitely. Please be patient. :)

**Comment by stefan-niedermann** [comment]
Another attempt to finally get this feature in: Everybody, **please help test** toggling checkboxes.

I attached a build of the branch #651 so you can easily install and test it:

[nextcloud-notes-toggle-checkboxes.apk.zip](https://github.com/stefan-niedermann/nextcloud-notes/files/4012345/nextcloud-notes-toggle-checkboxes.apk.zip)

I think, there might be one or two bugs left caused by RxMarkdown, which we didn't workaround yet, just post it in #651 and i will try to take care of them.

**Comment by panki27** [comment]
I just installed the build and tested it quickly. It seems to work fine on Android 9!
Although, I had to uninstall the FDroid version first (I guess because this is a debug version?).

The 3x2 single note widget doesn't seem to work, it just shows "Note not found (Notiz nicht gefunden)". Used to work before :( This might be a completely unrelated issue, though.

The only difference to the web version is that it doesn't toggle the strike-through effect, but thats okay imo.

Thanks a lot, great work! 👍

**Comment by stefan-niedermann** [comment]
> Although, I had to uninstall the FDroid version first (I guess because this is a debug version?).

That's right, because the app signature is different for a released version.

Regarding the widget: i think it's unrelated, for me it works, but we can have a look after this has been merged
Please test as many scenarios as you can imagine (only checkboxes, mixed with headlines, items containing markdown, only one checkbox in the whole note, ...)
